packages/react/src/components/DataTable/TableSelectRow.tsx

```diff
@@ -34,7 +85,8 @@ const TableSelectRow = ({
   };
   const InlineInputComponent = radio ? RadioButton : InlineCheckbox;
   const tableSelectRowClasses = classNames(`${prefix}--table-column-checkbox`, {
-    [className]: className,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    [className!]: className,
```

Comment:
think this way you can forego having to disable the eslint rule
```suggestion
    ...(className && {[className]: true}),
```

Reply:
Done, thank you!